sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala

```diff
@@ -86,6 +86,31 @@ trait CharVarcharTestSuite extends QueryTest with SQLTestUtils {
     }
   }
 
+  test("preserve char/varchar type info") {
+    Seq(CharType(5), VarcharType(5)).foreach { typ =>
+      withSQLConf(SQLConf.LEGACY_CHAR_VARCHAR_AS_STRING.key -> "true") {
```

Comment:
Could you also add the default behavior without setting any config. This IMO is a good thing to do, as that would make the test fail in future if some config default value is changed.